// pipedrive/dist/esm/versions/v1/models/mail-thread-all-of1.js
/* tslint:disable */
/* eslint-disable */
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
export const MailThreadAllOf1HasDraftFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
export const MailThreadAllOf1HasSentFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
export const MailThreadAllOf1ArchivedFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
export const MailThreadAllOf1SharedFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
export const MailThreadAllOf1ExternalDeletedFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
export const MailThreadAllOf1FirstMessageToMeFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
export const MailThreadAllOf1AllMessagesSentFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
