// pipedrive/dist/esm/versions/v1/models/mail-message-data.js
/* tslint:disable */
/* eslint-disable */
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
export const MailMessageDataMailTrackingStatusConst = {
    opened: 'opened',
    not_opened: 'not opened'
};
export const MailMessageDataMailLinkTrackingEnabledFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
export const MailMessageDataReadFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
export const MailMessageDataDraftFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
export const MailMessageDataSyncedFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
export const MailMessageDataDeletedFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
export const MailMessageDataHasBodyFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
export const MailMessageDataSentFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
export const MailMessageDataSentFromPipedriveFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
export const MailMessageDataSmartBccFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
export const MailMessageDataHasAttachmentsFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
export const MailMessageDataHasInlineAttachmentsFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
export const MailMessageDataHasRealAttachmentsFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
