// pipedrive/dist/esm/versions/v1/api/lead-labels-api.js
/* tslint:disable */
/* eslint-disable */
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
import globalAxios from 'axios';
// Some imports not used depending on template conditions
// @ts-ignore
import { DUMMY_BASE_URL, assertParamExists, setApiKeyToObject, setOAuthToObject, setSearchParams, serializeDataIfNeeded, toPathString, createRequestFunction } from '../common';
// @ts-ignore
import { BASE_PATH, BaseAPI } from '../base';
/**
 * LeadLabelsApi - axios parameter creator
 * @export
 */
export const LeadLabelsApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Creates a lead label.
         * @summary Add a lead label
         * @param {AddLeadLabelRequest} [AddLeadLabelRequest]

         * @throws {RequiredError}
         */
        addLeadLabel: (AddLeadLabelRequest) => __awaiter(this, void 0, void 0, function* () {
            const localVarPath = `/leadLabels`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign({ method: 'POST' }, baseOptions);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication api_key required
            yield setApiKeyToObject(localVarQueryParameter, "api_token", configuration);
            // authentication oauth2 required
            // oauth required
            yield setOAuthToObject(localVarHeaderParameter, "oauth2", ["leads:full"], configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions);
            localVarRequestOptions.data = serializeDataIfNeeded(AddLeadLabelRequest, localVarRequestOptions, configuration);
            return {
                url: toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         * Deletes a specific lead label.
         * @summary Delete a lead label
         * @param {string} id The ID of the lead label

         * @throws {RequiredError}
         */
        deleteLeadLabel: (id) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'id' is not null or undefined
            assertParamExists('deleteLeadLabel', 'id', id);
            const localVarPath = `/leadLabels/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign({ method: 'DELETE' }, baseOptions);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication api_key required
            yield setApiKeyToObject(localVarQueryParameter, "api_token", configuration);
            // authentication oauth2 required
            // oauth required
            yield setOAuthToObject(localVarHeaderParameter, "oauth2", ["leads:full"], configuration);
            setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions);
            return {
                url: toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         * Returns details of all lead labels. This endpoint does not support pagination and all labels are always returned.
         * @summary Get all lead labels

         * @throws {RequiredError}
         */
        getLeadLabels: () => __awaiter(this, void 0, void 0, function* () {
            const localVarPath = `/leadLabels`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign({ method: 'GET' }, baseOptions);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication api_key required
            yield setApiKeyToObject(localVarQueryParameter, "api_token", configuration);
            // authentication oauth2 required
            // oauth required
            yield setOAuthToObject(localVarHeaderParameter, "oauth2", ["leads:read", "leads:full"], configuration);
            setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions);
            return {
                url: toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         * Updates one or more properties of a lead label. Only properties included in the request will be updated.
         * @summary Update a lead label
         * @param {string} id The ID of the lead label
         * @param {UpdateLeadLabelRequest} [UpdateLeadLabelRequest]

         * @throws {RequiredError}
         */
        updateLeadLabel: (id, UpdateLeadLabelRequest) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'id' is not null or undefined
            assertParamExists('updateLeadLabel', 'id', id);
            const localVarPath = `/leadLabels/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign({ method: 'PATCH' }, baseOptions);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication api_key required
            yield setApiKeyToObject(localVarQueryParameter, "api_token", configuration);
            // authentication oauth2 required
            // oauth required
            yield setOAuthToObject(localVarHeaderParameter, "oauth2", ["leads:full"], configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions);
            localVarRequestOptions.data = serializeDataIfNeeded(UpdateLeadLabelRequest, localVarRequestOptions, configuration);
            return {
                url: toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
    };
};
/**
 * LeadLabelsApi - functional programming interface
 * @export
 */
export const LeadLabelsApiFp = function (configuration) {
    const localVarAxiosParamCreator = LeadLabelsApiAxiosParamCreator(configuration);
    return {
        /**
         * Creates a lead label.
         * @summary Add a lead label
         * @param {AddLeadLabelRequest} [AddLeadLabelRequest]

         * @throws {RequiredError}
         */
        addLeadLabel(AddLeadLabelRequest) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.addLeadLabel(AddLeadLabelRequest);
                return createRequestFunction(localVarAxiosArgs, globalAxios, BASE_PATH, configuration);
            });
        },
        /**
         * Deletes a specific lead label.
         * @summary Delete a lead label
         * @param {string} id The ID of the lead label

         * @throws {RequiredError}
         */
        deleteLeadLabel(id) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.deleteLeadLabel(id);
                return createRequestFunction(localVarAxiosArgs, globalAxios, BASE_PATH, configuration);
            });
        },
        /**
         * Returns details of all lead labels. This endpoint does not support pagination and all labels are always returned.
         * @summary Get all lead labels

         * @throws {RequiredError}
         */
        getLeadLabels() {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getLeadLabels();
                return createRequestFunction(localVarAxiosArgs, globalAxios, BASE_PATH, configuration);
            });
        },
        /**
         * Updates one or more properties of a lead label. Only properties included in the request will be updated.
         * @summary Update a lead label
         * @param {string} id The ID of the lead label
         * @param {UpdateLeadLabelRequest} [UpdateLeadLabelRequest]

         * @throws {RequiredError}
         */
        updateLeadLabel(id, UpdateLeadLabelRequest) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.updateLeadLabel(id, UpdateLeadLabelRequest);
                return createRequestFunction(localVarAxiosArgs, globalAxios, BASE_PATH, configuration);
            });
        },
    };
};
/**
 * LeadLabelsApi - factory interface
 * @export
 */
export const LeadLabelsApiFactory = function (configuration, basePath, axios) {
    const localVarFp = LeadLabelsApiFp(configuration);
    return {
        /**
         * Creates a lead label.
         * @summary Add a lead label
         * @param {LeadLabelsApiAddLeadLabelRequest} requestParameters Request parameters.

         * @throws {RequiredError}
         */
        addLeadLabel(requestParameters = {}) {
            return localVarFp.addLeadLabel(requestParameters.AddLeadLabelRequest).then((request) => request(axios, basePath));
        },
        /**
         * Deletes a specific lead label.
         * @summary Delete a lead label
         * @param {LeadLabelsApiDeleteLeadLabelRequest} requestParameters Request parameters.

         * @throws {RequiredError}
         */
        deleteLeadLabel(requestParameters) {
            return localVarFp.deleteLeadLabel(requestParameters.id).then((request) => request(axios, basePath));
        },
        /**
         * Returns details of all lead labels. This endpoint does not support pagination and all labels are always returned.
         * @summary Get all lead labels

         * @throws {RequiredError}
         */
        getLeadLabels() {
            return localVarFp.getLeadLabels().then((request) => request(axios, basePath));
        },
        /**
         * Updates one or more properties of a lead label. Only properties included in the request will be updated.
         * @summary Update a lead label
         * @param {LeadLabelsApiUpdateLeadLabelRequest} requestParameters Request parameters.

         * @throws {RequiredError}
         */
        updateLeadLabel(requestParameters) {
            return localVarFp.updateLeadLabel(requestParameters.id, requestParameters.UpdateLeadLabelRequest).then((request) => request(axios, basePath));
        },
    };
};
/**
 * LeadLabelsApi - object-oriented interface
 * @export
 * @class LeadLabelsApi
 * @extends {BaseAPI}
 */
export class LeadLabelsApi extends BaseAPI {
    /**
     * Creates a lead label.
     * @summary Add a lead label
     * @param {LeadLabelsApiAddLeadLabelRequest} requestParameters Request parameters.

     * @throws {RequiredError}
     * @memberof LeadLabelsApi
     */
    addLeadLabel(requestParameters = {}) {
        return LeadLabelsApiFp(this.configuration).addLeadLabel(requestParameters.AddLeadLabelRequest).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Deletes a specific lead label.
     * @summary Delete a lead label
     * @param {LeadLabelsApiDeleteLeadLabelRequest} requestParameters Request parameters.

     * @throws {RequiredError}
     * @memberof LeadLabelsApi
     */
    deleteLeadLabel(requestParameters) {
        return LeadLabelsApiFp(this.configuration).deleteLeadLabel(requestParameters.id).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Returns details of all lead labels. This endpoint does not support pagination and all labels are always returned.
     * @summary Get all lead labels

     * @throws {RequiredError}
     * @memberof LeadLabelsApi
     */
    getLeadLabels() {
        return LeadLabelsApiFp(this.configuration).getLeadLabels().then((request) => request(this.axios, this.basePath));
    }
    /**
     * Updates one or more properties of a lead label. Only properties included in the request will be updated.
     * @summary Update a lead label
     * @param {LeadLabelsApiUpdateLeadLabelRequest} requestParameters Request parameters.

     * @throws {RequiredError}
     * @memberof LeadLabelsApi
     */
    updateLeadLabel(requestParameters) {
        return LeadLabelsApiFp(this.configuration).updateLeadLabel(requestParameters.id, requestParameters.UpdateLeadLabelRequest).then((request) => request(this.axios, this.basePath));
    }
}
