// pipedrive/dist/esm/versions/v1/api/meetings-api.js
/* tslint:disable */
/* eslint-disable */
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
import globalAxios from 'axios';
// Some imports not used depending on template conditions
// @ts-ignore
import { DUMMY_BASE_URL, assertParamExists, setApiKeyToObject, setOAuthToObject, setSearchParams, serializeDataIfNeeded, toPathString, createRequestFunction } from '../common';
// @ts-ignore
import { BASE_PATH, BaseAPI } from '../base';
/**
 * MeetingsApi - axios parameter creator
 * @export
 */
export const MeetingsApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * A video calling provider must call this endpoint to remove the link between a user and the installed video calling app.
         * @summary Delete the link between a user and the installed video call integration
         * @param {string} id Unique identifier linking a user to the installed integration

         * @throws {RequiredError}
         */
        deleteUserProviderLink: (id) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'id' is not null or undefined
            assertParamExists('deleteUserProviderLink', 'id', id);
            const localVarPath = `/meetings/userProviderLinks/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign({ method: 'DELETE' }, baseOptions);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication api_key required
            yield setApiKeyToObject(localVarQueryParameter, "api_token", configuration);
            // authentication oauth2 required
            // oauth required
            yield setOAuthToObject(localVarHeaderParameter, "oauth2", ["video-calls"], configuration);
            setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions);
            return {
                url: toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         * A video calling provider must call this endpoint after a user has installed the video calling app so that the new user\'s information is sent.
         * @summary Link a user with the installed video call integration
         * @param {AddUserProviderLinkRequest} [AddUserProviderLinkRequest]

         * @throws {RequiredError}
         */
        saveUserProviderLink: (AddUserProviderLinkRequest) => __awaiter(this, void 0, void 0, function* () {
            const localVarPath = `/meetings/userProviderLinks`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign({ method: 'POST' }, baseOptions);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication api_key required
            yield setApiKeyToObject(localVarQueryParameter, "api_token", configuration);
            // authentication oauth2 required
            // oauth required
            yield setOAuthToObject(localVarHeaderParameter, "oauth2", ["video-calls"], configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions);
            localVarRequestOptions.data = serializeDataIfNeeded(AddUserProviderLinkRequest, localVarRequestOptions, configuration);
            return {
                url: toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
    };
};
/**
 * MeetingsApi - functional programming interface
 * @export
 */
export const MeetingsApiFp = function (configuration) {
    const localVarAxiosParamCreator = MeetingsApiAxiosParamCreator(configuration);
    return {
        /**
         * A video calling provider must call this endpoint to remove the link between a user and the installed video calling app.
         * @summary Delete the link between a user and the installed video call integration
         * @param {string} id Unique identifier linking a user to the installed integration

         * @throws {RequiredError}
         */
        deleteUserProviderLink(id) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.deleteUserProviderLink(id);
                return createRequestFunction(localVarAxiosArgs, globalAxios, BASE_PATH, configuration);
            });
        },
        /**
         * A video calling provider must call this endpoint after a user has installed the video calling app so that the new user\'s information is sent.
         * @summary Link a user with the installed video call integration
         * @param {AddUserProviderLinkRequest} [AddUserProviderLinkRequest]

         * @throws {RequiredError}
         */
        saveUserProviderLink(AddUserProviderLinkRequest) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.saveUserProviderLink(AddUserProviderLinkRequest);
                return createRequestFunction(localVarAxiosArgs, globalAxios, BASE_PATH, configuration);
            });
        },
    };
};
/**
 * MeetingsApi - factory interface
 * @export
 */
export const MeetingsApiFactory = function (configuration, basePath, axios) {
    const localVarFp = MeetingsApiFp(configuration);
    return {
        /**
         * A video calling provider must call this endpoint to remove the link between a user and the installed video calling app.
         * @summary Delete the link between a user and the installed video call integration
         * @param {MeetingsApiDeleteUserProviderLinkRequest} requestParameters Request parameters.

         * @throws {RequiredError}
         */
        deleteUserProviderLink(requestParameters) {
            return localVarFp.deleteUserProviderLink(requestParameters.id).then((request) => request(axios, basePath));
        },
        /**
         * A video calling provider must call this endpoint after a user has installed the video calling app so that the new user\'s information is sent.
         * @summary Link a user with the installed video call integration
         * @param {MeetingsApiSaveUserProviderLinkRequest} requestParameters Request parameters.

         * @throws {RequiredError}
         */
        saveUserProviderLink(requestParameters = {}) {
            return localVarFp.saveUserProviderLink(requestParameters.AddUserProviderLinkRequest).then((request) => request(axios, basePath));
        },
    };
};
/**
 * MeetingsApi - object-oriented interface
 * @export
 * @class MeetingsApi
 * @extends {BaseAPI}
 */
export class MeetingsApi extends BaseAPI {
    /**
     * A video calling provider must call this endpoint to remove the link between a user and the installed video calling app.
     * @summary Delete the link between a user and the installed video call integration
     * @param {MeetingsApiDeleteUserProviderLinkRequest} requestParameters Request parameters.

     * @throws {RequiredError}
     * @memberof MeetingsApi
     */
    deleteUserProviderLink(requestParameters) {
        return MeetingsApiFp(this.configuration).deleteUserProviderLink(requestParameters.id).then((request) => request(this.axios, this.basePath));
    }
    /**
     * A video calling provider must call this endpoint after a user has installed the video calling app so that the new user\'s information is sent.
     * @summary Link a user with the installed video call integration
     * @param {MeetingsApiSaveUserProviderLinkRequest} requestParameters Request parameters.

     * @throws {RequiredError}
     * @memberof MeetingsApi
     */
    saveUserProviderLink(requestParameters = {}) {
        return MeetingsApiFp(this.configuration).saveUserProviderLink(requestParameters.AddUserProviderLinkRequest).then((request) => request(this.axios, this.basePath));
    }
}
