// pipedrive/dist/versions/v1/api/lead-sources-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LeadSourcesApi = exports.LeadSourcesApiFactory = exports.LeadSourcesApiFp = exports.LeadSourcesApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * LeadSourcesApi - axios parameter creator
 * @export
 */
const LeadSourcesApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Returns all lead sources. Please note that the list of lead sources is fixed, it cannot be modified. All leads created through the Pipedrive API will have a lead source `API` assigned.
         * @summary Get all lead sources

         * @throws {RequiredError}
         */
        getLeadSources: () => __awaiter(this, void 0, void 0, function* () {
            const localVarPath = `/leadSources`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign({ method: 'GET' }, baseOptions);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication api_key required
            yield (0, common_1.setApiKeyToObject)(localVarQueryParameter, "api_token", configuration);
            // authentication oauth2 required
            // oauth required
            yield (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["leads:read", "leads:full"], configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
    };
};
exports.LeadSourcesApiAxiosParamCreator = LeadSourcesApiAxiosParamCreator;
/**
 * LeadSourcesApi - functional programming interface
 * @export
 */
const LeadSourcesApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.LeadSourcesApiAxiosParamCreator)(configuration);
    return {
        /**
         * Returns all lead sources. Please note that the list of lead sources is fixed, it cannot be modified. All leads created through the Pipedrive API will have a lead source `API` assigned.
         * @summary Get all lead sources

         * @throws {RequiredError}
         */
        getLeadSources() {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getLeadSources();
                return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
    };
};
exports.LeadSourcesApiFp = LeadSourcesApiFp;
/**
 * LeadSourcesApi - factory interface
 * @export
 */
const LeadSourcesApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.LeadSourcesApiFp)(configuration);
    return {
        /**
         * Returns all lead sources. Please note that the list of lead sources is fixed, it cannot be modified. All leads created through the Pipedrive API will have a lead source `API` assigned.
         * @summary Get all lead sources

         * @throws {RequiredError}
         */
        getLeadSources() {
            return localVarFp.getLeadSources().then((request) => request(axios, basePath));
        },
    };
};
exports.LeadSourcesApiFactory = LeadSourcesApiFactory;
/**
 * LeadSourcesApi - object-oriented interface
 * @export
 * @class LeadSourcesApi
 * @extends {BaseAPI}
 */
class LeadSourcesApi extends base_1.BaseAPI {
    /**
     * Returns all lead sources. Please note that the list of lead sources is fixed, it cannot be modified. All leads created through the Pipedrive API will have a lead source `API` assigned.
     * @summary Get all lead sources

     * @throws {RequiredError}
     * @memberof LeadSourcesApi
     */
    getLeadSources() {
        return (0, exports.LeadSourcesApiFp)(this.configuration).getLeadSources().then((request) => request(this.axios, this.basePath));
    }
}
exports.LeadSourcesApi = LeadSourcesApi;
