// pipedrive/dist/versions/v1/api/organization-fields-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrganizationFieldsApi = exports.OrganizationFieldsApiFactory = exports.OrganizationFieldsApiFp = exports.OrganizationFieldsApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * OrganizationFieldsApi - axios parameter creator
 * @export
 */
const OrganizationFieldsApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Adds a new organization field. For more information, see the tutorial for <a href=\"https://pipedrive.readme.io/docs/adding-a-new-custom-field\" target=\"_blank\" rel=\"noopener noreferrer\">adding a new custom field</a>.
         * @summary Add a new organization field
         * @param {CreateFieldRequest} [CreateFieldRequest]

         * @throws {RequiredError}
         */
        addOrganizationField: (CreateFieldRequest) => __awaiter(this, void 0, void 0, function* () {
            const localVarPath = `/organizationFields`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign({ method: 'POST' }, baseOptions);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication api_key required
            yield (0, common_1.setApiKeyToObject)(localVarQueryParameter, "api_token", configuration);
            // authentication oauth2 required
            // oauth required
            yield (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["admin"], configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(CreateFieldRequest, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         * Marks a field as deleted. For more information, see the tutorial for <a href=\"https://pipedrive.readme.io/docs/deleting-a-custom-field\" target=\"_blank\" rel=\"noopener noreferrer\">deleting a custom field</a>.
         * @summary Delete an organization field
         * @param {number} id The ID of the field

         * @throws {RequiredError}
         */
        deleteOrganizationField: (id) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('deleteOrganizationField', 'id', id);
            const localVarPath = `/organizationFields/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign({ method: 'DELETE' }, baseOptions);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication api_key required
            yield (0, common_1.setApiKeyToObject)(localVarQueryParameter, "api_token", configuration);
            // authentication oauth2 required
            // oauth required
            yield (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["admin"], configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         * Marks multiple fields as deleted.
         * @summary Delete multiple organization fields in bulk
         * @param {string} ids The comma-separated field IDs to delete

         * @throws {RequiredError}
         */
        deleteOrganizationFields: (ids) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'ids' is not null or undefined
            (0, common_1.assertParamExists)('deleteOrganizationFields', 'ids', ids);
            const localVarPath = `/organizationFields`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign({ method: 'DELETE' }, baseOptions);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication api_key required
            yield (0, common_1.setApiKeyToObject)(localVarQueryParameter, "api_token", configuration);
            // authentication oauth2 required
            // oauth required
            yield (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["admin"], configuration);
            if (ids !== undefined) {
                localVarQueryParameter['ids'] = ids;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         * Returns data about a specific organization field.
         * @summary Get one organization field
         * @param {number} id The ID of the field

         * @throws {RequiredError}
         */
        getOrganizationField: (id) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('getOrganizationField', 'id', id);
            const localVarPath = `/organizationFields/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign({ method: 'GET' }, baseOptions);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication api_key required
            yield (0, common_1.setApiKeyToObject)(localVarQueryParameter, "api_token", configuration);
            // authentication oauth2 required
            // oauth required
            yield (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["contacts:read", "contacts:full", "admin"], configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         * Returns data about all organization fields.
         * @summary Get all organization fields
         * @param {number} [start] Pagination start
         * @param {number} [limit] Items shown per page

         * @throws {RequiredError}
         */
        getOrganizationFields: (start, limit) => __awaiter(this, void 0, void 0, function* () {
            const localVarPath = `/organizationFields`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign({ method: 'GET' }, baseOptions);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication api_key required
            yield (0, common_1.setApiKeyToObject)(localVarQueryParameter, "api_token", configuration);
            // authentication oauth2 required
            // oauth required
            yield (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["contacts:read", "contacts:full", "admin"], configuration);
            if (start !== undefined) {
                localVarQueryParameter['start'] = start;
            }
            if (limit !== undefined) {
                localVarQueryParameter['limit'] = limit;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         * Updates an organization field. For more information, see the tutorial for <a href=\" https://pipedrive.readme.io/docs/updating-custom-field-value \" target=\"_blank\" rel=\"noopener noreferrer\">updating custom fields\' values</a>.
         * @summary Update an organization field
         * @param {number} id The ID of the field
         * @param {UpdateFieldRequest} [UpdateFieldRequest]

         * @throws {RequiredError}
         */
        updateOrganizationField: (id, UpdateFieldRequest) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('updateOrganizationField', 'id', id);
            const localVarPath = `/organizationFields/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign({ method: 'PUT' }, baseOptions);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication api_key required
            yield (0, common_1.setApiKeyToObject)(localVarQueryParameter, "api_token", configuration);
            // authentication oauth2 required
            // oauth required
            yield (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["admin"], configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(UpdateFieldRequest, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
    };
};
exports.OrganizationFieldsApiAxiosParamCreator = OrganizationFieldsApiAxiosParamCreator;
/**
 * OrganizationFieldsApi - functional programming interface
 * @export
 */
const OrganizationFieldsApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.OrganizationFieldsApiAxiosParamCreator)(configuration);
    return {
        /**
         * Adds a new organization field. For more information, see the tutorial for <a href=\"https://pipedrive.readme.io/docs/adding-a-new-custom-field\" target=\"_blank\" rel=\"noopener noreferrer\">adding a new custom field</a>.
         * @summary Add a new organization field
         * @param {CreateFieldRequest} [CreateFieldRequest]

         * @throws {RequiredError}
         */
        addOrganizationField(CreateFieldRequest) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.addOrganizationField(CreateFieldRequest);
                return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         * Marks a field as deleted. For more information, see the tutorial for <a href=\"https://pipedrive.readme.io/docs/deleting-a-custom-field\" target=\"_blank\" rel=\"noopener noreferrer\">deleting a custom field</a>.
         * @summary Delete an organization field
         * @param {number} id The ID of the field

         * @throws {RequiredError}
         */
        deleteOrganizationField(id) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.deleteOrganizationField(id);
                return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         * Marks multiple fields as deleted.
         * @summary Delete multiple organization fields in bulk
         * @param {string} ids The comma-separated field IDs to delete

         * @throws {RequiredError}
         */
        deleteOrganizationFields(ids) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.deleteOrganizationFields(ids);
                return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         * Returns data about a specific organization field.
         * @summary Get one organization field
         * @param {number} id The ID of the field

         * @throws {RequiredError}
         */
        getOrganizationField(id) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getOrganizationField(id);
                return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         * Returns data about all organization fields.
         * @summary Get all organization fields
         * @param {number} [start] Pagination start
         * @param {number} [limit] Items shown per page

         * @throws {RequiredError}
         */
        getOrganizationFields(start, limit) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getOrganizationFields(start, limit);
                return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         * Updates an organization field. For more information, see the tutorial for <a href=\" https://pipedrive.readme.io/docs/updating-custom-field-value \" target=\"_blank\" rel=\"noopener noreferrer\">updating custom fields\' values</a>.
         * @summary Update an organization field
         * @param {number} id The ID of the field
         * @param {UpdateFieldRequest} [UpdateFieldRequest]

         * @throws {RequiredError}
         */
        updateOrganizationField(id, UpdateFieldRequest) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.updateOrganizationField(id, UpdateFieldRequest);
                return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
    };
};
exports.OrganizationFieldsApiFp = OrganizationFieldsApiFp;
/**
 * OrganizationFieldsApi - factory interface
 * @export
 */
const OrganizationFieldsApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.OrganizationFieldsApiFp)(configuration);
    return {
        /**
         * Adds a new organization field. For more information, see the tutorial for <a href=\"https://pipedrive.readme.io/docs/adding-a-new-custom-field\" target=\"_blank\" rel=\"noopener noreferrer\">adding a new custom field</a>.
         * @summary Add a new organization field
         * @param {OrganizationFieldsApiAddOrganizationFieldRequest} requestParameters Request parameters.

         * @throws {RequiredError}
         */
        addOrganizationField(requestParameters = {}) {
            return localVarFp.addOrganizationField(requestParameters.CreateFieldRequest).then((request) => request(axios, basePath));
        },
        /**
         * Marks a field as deleted. For more information, see the tutorial for <a href=\"https://pipedrive.readme.io/docs/deleting-a-custom-field\" target=\"_blank\" rel=\"noopener noreferrer\">deleting a custom field</a>.
         * @summary Delete an organization field
         * @param {OrganizationFieldsApiDeleteOrganizationFieldRequest} requestParameters Request parameters.

         * @throws {RequiredError}
         */
        deleteOrganizationField(requestParameters) {
            return localVarFp.deleteOrganizationField(requestParameters.id).then((request) => request(axios, basePath));
        },
        /**
         * Marks multiple fields as deleted.
         * @summary Delete multiple organization fields in bulk
         * @param {OrganizationFieldsApiDeleteOrganizationFieldsRequest} requestParameters Request parameters.

         * @throws {RequiredError}
         */
        deleteOrganizationFields(requestParameters) {
            return localVarFp.deleteOrganizationFields(requestParameters.ids).then((request) => request(axios, basePath));
        },
        /**
         * Returns data about a specific organization field.
         * @summary Get one organization field
         * @param {OrganizationFieldsApiGetOrganizationFieldRequest} requestParameters Request parameters.

         * @throws {RequiredError}
         */
        getOrganizationField(requestParameters) {
            return localVarFp.getOrganizationField(requestParameters.id).then((request) => request(axios, basePath));
        },
        /**
         * Returns data about all organization fields.
         * @summary Get all organization fields
         * @param {OrganizationFieldsApiGetOrganizationFieldsRequest} requestParameters Request parameters.

         * @throws {RequiredError}
         */
        getOrganizationFields(requestParameters = {}) {
            return localVarFp.getOrganizationFields(requestParameters.start, requestParameters.limit).then((request) => request(axios, basePath));
        },
        /**
         * Updates an organization field. For more information, see the tutorial for <a href=\" https://pipedrive.readme.io/docs/updating-custom-field-value \" target=\"_blank\" rel=\"noopener noreferrer\">updating custom fields\' values</a>.
         * @summary Update an organization field
         * @param {OrganizationFieldsApiUpdateOrganizationFieldRequest} requestParameters Request parameters.

         * @throws {RequiredError}
         */
        updateOrganizationField(requestParameters) {
            return localVarFp.updateOrganizationField(requestParameters.id, requestParameters.UpdateFieldRequest).then((request) => request(axios, basePath));
        },
    };
};
exports.OrganizationFieldsApiFactory = OrganizationFieldsApiFactory;
/**
 * OrganizationFieldsApi - object-oriented interface
 * @export
 * @class OrganizationFieldsApi
 * @extends {BaseAPI}
 */
class OrganizationFieldsApi extends base_1.BaseAPI {
    /**
     * Adds a new organization field. For more information, see the tutorial for <a href=\"https://pipedrive.readme.io/docs/adding-a-new-custom-field\" target=\"_blank\" rel=\"noopener noreferrer\">adding a new custom field</a>.
     * @summary Add a new organization field
     * @param {OrganizationFieldsApiAddOrganizationFieldRequest} requestParameters Request parameters.

     * @throws {RequiredError}
     * @memberof OrganizationFieldsApi
     */
    addOrganizationField(requestParameters = {}) {
        return (0, exports.OrganizationFieldsApiFp)(this.configuration).addOrganizationField(requestParameters.CreateFieldRequest).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Marks a field as deleted. For more information, see the tutorial for <a href=\"https://pipedrive.readme.io/docs/deleting-a-custom-field\" target=\"_blank\" rel=\"noopener noreferrer\">deleting a custom field</a>.
     * @summary Delete an organization field
     * @param {OrganizationFieldsApiDeleteOrganizationFieldRequest} requestParameters Request parameters.

     * @throws {RequiredError}
     * @memberof OrganizationFieldsApi
     */
    deleteOrganizationField(requestParameters) {
        return (0, exports.OrganizationFieldsApiFp)(this.configuration).deleteOrganizationField(requestParameters.id).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Marks multiple fields as deleted.
     * @summary Delete multiple organization fields in bulk
     * @param {OrganizationFieldsApiDeleteOrganizationFieldsRequest} requestParameters Request parameters.

     * @throws {RequiredError}
     * @memberof OrganizationFieldsApi
     */
    deleteOrganizationFields(requestParameters) {
        return (0, exports.OrganizationFieldsApiFp)(this.configuration).deleteOrganizationFields(requestParameters.ids).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Returns data about a specific organization field.
     * @summary Get one organization field
     * @param {OrganizationFieldsApiGetOrganizationFieldRequest} requestParameters Request parameters.

     * @throws {RequiredError}
     * @memberof OrganizationFieldsApi
     */
    getOrganizationField(requestParameters) {
        return (0, exports.OrganizationFieldsApiFp)(this.configuration).getOrganizationField(requestParameters.id).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Returns data about all organization fields.
     * @summary Get all organization fields
     * @param {OrganizationFieldsApiGetOrganizationFieldsRequest} requestParameters Request parameters.

     * @throws {RequiredError}
     * @memberof OrganizationFieldsApi
     */
    getOrganizationFields(requestParameters = {}) {
        return (0, exports.OrganizationFieldsApiFp)(this.configuration).getOrganizationFields(requestParameters.start, requestParameters.limit).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Updates an organization field. For more information, see the tutorial for <a href=\" https://pipedrive.readme.io/docs/updating-custom-field-value \" target=\"_blank\" rel=\"noopener noreferrer\">updating custom fields\' values</a>.
     * @summary Update an organization field
     * @param {OrganizationFieldsApiUpdateOrganizationFieldRequest} requestParameters Request parameters.

     * @throws {RequiredError}
     * @memberof OrganizationFieldsApi
     */
    updateOrganizationField(requestParameters) {
        return (0, exports.OrganizationFieldsApiFp)(this.configuration).updateOrganizationField(requestParameters.id, requestParameters.UpdateFieldRequest).then((request) => request(this.axios, this.basePath));
    }
}
exports.OrganizationFieldsApi = OrganizationFieldsApi;
