// pipedrive/dist/versions/v1/api/person-fields-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PersonFieldsApi = exports.PersonFieldsApiFactory = exports.PersonFieldsApiFp = exports.PersonFieldsApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * PersonFieldsApi - axios parameter creator
 * @export
 */
const PersonFieldsApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Adds a new person field. For more information, see the tutorial for <a href=\"https://pipedrive.readme.io/docs/adding-a-new-custom-field\" target=\"_blank\" rel=\"noopener noreferrer\">adding a new custom field</a>.
         * @summary Add a new person field
         * @param {CreateFieldRequest} [CreateFieldRequest]

         * @throws {RequiredError}
         */
        addPersonField: (CreateFieldRequest) => __awaiter(this, void 0, void 0, function* () {
            const localVarPath = `/personFields`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign({ method: 'POST' }, baseOptions);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication api_key required
            yield (0, common_1.setApiKeyToObject)(localVarQueryParameter, "api_token", configuration);
            // authentication oauth2 required
            // oauth required
            yield (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["admin"], configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(CreateFieldRequest, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         * Marks a field as deleted. For more information, see the tutorial for <a href=\"https://pipedrive.readme.io/docs/deleting-a-custom-field\" target=\"_blank\" rel=\"noopener noreferrer\">deleting a custom field</a>.
         * @summary Delete a person field
         * @param {number} id The ID of the field

         * @throws {RequiredError}
         */
        deletePersonField: (id) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('deletePersonField', 'id', id);
            const localVarPath = `/personFields/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign({ method: 'DELETE' }, baseOptions);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication api_key required
            yield (0, common_1.setApiKeyToObject)(localVarQueryParameter, "api_token", configuration);
            // authentication oauth2 required
            // oauth required
            yield (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["admin"], configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         * Marks multiple fields as deleted.
         * @summary Delete multiple person fields in bulk
         * @param {string} ids The comma-separated field IDs to delete

         * @throws {RequiredError}
         */
        deletePersonFields: (ids) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'ids' is not null or undefined
            (0, common_1.assertParamExists)('deletePersonFields', 'ids', ids);
            const localVarPath = `/personFields`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign({ method: 'DELETE' }, baseOptions);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication api_key required
            yield (0, common_1.setApiKeyToObject)(localVarQueryParameter, "api_token", configuration);
            // authentication oauth2 required
            // oauth required
            yield (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["admin"], configuration);
            if (ids !== undefined) {
                localVarQueryParameter['ids'] = ids;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         * Returns data about a specific person field.
         * @summary Get one person field
         * @param {number} id The ID of the field

         * @throws {RequiredError}
         */
        getPersonField: (id) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('getPersonField', 'id', id);
            const localVarPath = `/personFields/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign({ method: 'GET' }, baseOptions);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication api_key required
            yield (0, common_1.setApiKeyToObject)(localVarQueryParameter, "api_token", configuration);
            // authentication oauth2 required
            // oauth required
            yield (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["contacts:read", "contacts:full", "admin"], configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         * Returns data about all person fields.<br>If a company uses the [Campaigns product](https://pipedrive.readme.io/docs/campaigns-in-pipedrive-api), then this endpoint will also return the `data.marketing_status` field.
         * @summary Get all person fields
         * @param {number} [start] Pagination start
         * @param {number} [limit] Items shown per page

         * @throws {RequiredError}
         */
        getPersonFields: (start, limit) => __awaiter(this, void 0, void 0, function* () {
            const localVarPath = `/personFields`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign({ method: 'GET' }, baseOptions);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication api_key required
            yield (0, common_1.setApiKeyToObject)(localVarQueryParameter, "api_token", configuration);
            // authentication oauth2 required
            // oauth required
            yield (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["contacts:read", "contacts:full", "admin"], configuration);
            if (start !== undefined) {
                localVarQueryParameter['start'] = start;
            }
            if (limit !== undefined) {
                localVarQueryParameter['limit'] = limit;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         * Updates a person field. For more information, see the tutorial for <a href=\" https://pipedrive.readme.io/docs/updating-custom-field-value \" target=\"_blank\" rel=\"noopener noreferrer\">updating custom fields\' values</a>.
         * @summary Update a person field
         * @param {number} id The ID of the field
         * @param {UpdateFieldRequest} [UpdateFieldRequest]

         * @throws {RequiredError}
         */
        updatePersonField: (id, UpdateFieldRequest) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('updatePersonField', 'id', id);
            const localVarPath = `/personFields/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign({ method: 'PUT' }, baseOptions);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication api_key required
            yield (0, common_1.setApiKeyToObject)(localVarQueryParameter, "api_token", configuration);
            // authentication oauth2 required
            // oauth required
            yield (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["admin"], configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(UpdateFieldRequest, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
    };
};
exports.PersonFieldsApiAxiosParamCreator = PersonFieldsApiAxiosParamCreator;
/**
 * PersonFieldsApi - functional programming interface
 * @export
 */
const PersonFieldsApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.PersonFieldsApiAxiosParamCreator)(configuration);
    return {
        /**
         * Adds a new person field. For more information, see the tutorial for <a href=\"https://pipedrive.readme.io/docs/adding-a-new-custom-field\" target=\"_blank\" rel=\"noopener noreferrer\">adding a new custom field</a>.
         * @summary Add a new person field
         * @param {CreateFieldRequest} [CreateFieldRequest]

         * @throws {RequiredError}
         */
        addPersonField(CreateFieldRequest) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.addPersonField(CreateFieldRequest);
                return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         * Marks a field as deleted. For more information, see the tutorial for <a href=\"https://pipedrive.readme.io/docs/deleting-a-custom-field\" target=\"_blank\" rel=\"noopener noreferrer\">deleting a custom field</a>.
         * @summary Delete a person field
         * @param {number} id The ID of the field

         * @throws {RequiredError}
         */
        deletePersonField(id) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.deletePersonField(id);
                return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         * Marks multiple fields as deleted.
         * @summary Delete multiple person fields in bulk
         * @param {string} ids The comma-separated field IDs to delete

         * @throws {RequiredError}
         */
        deletePersonFields(ids) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.deletePersonFields(ids);
                return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         * Returns data about a specific person field.
         * @summary Get one person field
         * @param {number} id The ID of the field

         * @throws {RequiredError}
         */
        getPersonField(id) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getPersonField(id);
                return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         * Returns data about all person fields.<br>If a company uses the [Campaigns product](https://pipedrive.readme.io/docs/campaigns-in-pipedrive-api), then this endpoint will also return the `data.marketing_status` field.
         * @summary Get all person fields
         * @param {number} [start] Pagination start
         * @param {number} [limit] Items shown per page

         * @throws {RequiredError}
         */
        getPersonFields(start, limit) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getPersonFields(start, limit);
                return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         * Updates a person field. For more information, see the tutorial for <a href=\" https://pipedrive.readme.io/docs/updating-custom-field-value \" target=\"_blank\" rel=\"noopener noreferrer\">updating custom fields\' values</a>.
         * @summary Update a person field
         * @param {number} id The ID of the field
         * @param {UpdateFieldRequest} [UpdateFieldRequest]

         * @throws {RequiredError}
         */
        updatePersonField(id, UpdateFieldRequest) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.updatePersonField(id, UpdateFieldRequest);
                return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
    };
};
exports.PersonFieldsApiFp = PersonFieldsApiFp;
/**
 * PersonFieldsApi - factory interface
 * @export
 */
const PersonFieldsApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.PersonFieldsApiFp)(configuration);
    return {
        /**
         * Adds a new person field. For more information, see the tutorial for <a href=\"https://pipedrive.readme.io/docs/adding-a-new-custom-field\" target=\"_blank\" rel=\"noopener noreferrer\">adding a new custom field</a>.
         * @summary Add a new person field
         * @param {PersonFieldsApiAddPersonFieldRequest} requestParameters Request parameters.

         * @throws {RequiredError}
         */
        addPersonField(requestParameters = {}) {
            return localVarFp.addPersonField(requestParameters.CreateFieldRequest).then((request) => request(axios, basePath));
        },
        /**
         * Marks a field as deleted. For more information, see the tutorial for <a href=\"https://pipedrive.readme.io/docs/deleting-a-custom-field\" target=\"_blank\" rel=\"noopener noreferrer\">deleting a custom field</a>.
         * @summary Delete a person field
         * @param {PersonFieldsApiDeletePersonFieldRequest} requestParameters Request parameters.

         * @throws {RequiredError}
         */
        deletePersonField(requestParameters) {
            return localVarFp.deletePersonField(requestParameters.id).then((request) => request(axios, basePath));
        },
        /**
         * Marks multiple fields as deleted.
         * @summary Delete multiple person fields in bulk
         * @param {PersonFieldsApiDeletePersonFieldsRequest} requestParameters Request parameters.

         * @throws {RequiredError}
         */
        deletePersonFields(requestParameters) {
            return localVarFp.deletePersonFields(requestParameters.ids).then((request) => request(axios, basePath));
        },
        /**
         * Returns data about a specific person field.
         * @summary Get one person field
         * @param {PersonFieldsApiGetPersonFieldRequest} requestParameters Request parameters.

         * @throws {RequiredError}
         */
        getPersonField(requestParameters) {
            return localVarFp.getPersonField(requestParameters.id).then((request) => request(axios, basePath));
        },
        /**
         * Returns data about all person fields.<br>If a company uses the [Campaigns product](https://pipedrive.readme.io/docs/campaigns-in-pipedrive-api), then this endpoint will also return the `data.marketing_status` field.
         * @summary Get all person fields
         * @param {PersonFieldsApiGetPersonFieldsRequest} requestParameters Request parameters.

         * @throws {RequiredError}
         */
        getPersonFields(requestParameters = {}) {
            return localVarFp.getPersonFields(requestParameters.start, requestParameters.limit).then((request) => request(axios, basePath));
        },
        /**
         * Updates a person field. For more information, see the tutorial for <a href=\" https://pipedrive.readme.io/docs/updating-custom-field-value \" target=\"_blank\" rel=\"noopener noreferrer\">updating custom fields\' values</a>.
         * @summary Update a person field
         * @param {PersonFieldsApiUpdatePersonFieldRequest} requestParameters Request parameters.

         * @throws {RequiredError}
         */
        updatePersonField(requestParameters) {
            return localVarFp.updatePersonField(requestParameters.id, requestParameters.UpdateFieldRequest).then((request) => request(axios, basePath));
        },
    };
};
exports.PersonFieldsApiFactory = PersonFieldsApiFactory;
/**
 * PersonFieldsApi - object-oriented interface
 * @export
 * @class PersonFieldsApi
 * @extends {BaseAPI}
 */
class PersonFieldsApi extends base_1.BaseAPI {
    /**
     * Adds a new person field. For more information, see the tutorial for <a href=\"https://pipedrive.readme.io/docs/adding-a-new-custom-field\" target=\"_blank\" rel=\"noopener noreferrer\">adding a new custom field</a>.
     * @summary Add a new person field
     * @param {PersonFieldsApiAddPersonFieldRequest} requestParameters Request parameters.

     * @throws {RequiredError}
     * @memberof PersonFieldsApi
     */
    addPersonField(requestParameters = {}) {
        return (0, exports.PersonFieldsApiFp)(this.configuration).addPersonField(requestParameters.CreateFieldRequest).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Marks a field as deleted. For more information, see the tutorial for <a href=\"https://pipedrive.readme.io/docs/deleting-a-custom-field\" target=\"_blank\" rel=\"noopener noreferrer\">deleting a custom field</a>.
     * @summary Delete a person field
     * @param {PersonFieldsApiDeletePersonFieldRequest} requestParameters Request parameters.

     * @throws {RequiredError}
     * @memberof PersonFieldsApi
     */
    deletePersonField(requestParameters) {
        return (0, exports.PersonFieldsApiFp)(this.configuration).deletePersonField(requestParameters.id).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Marks multiple fields as deleted.
     * @summary Delete multiple person fields in bulk
     * @param {PersonFieldsApiDeletePersonFieldsRequest} requestParameters Request parameters.

     * @throws {RequiredError}
     * @memberof PersonFieldsApi
     */
    deletePersonFields(requestParameters) {
        return (0, exports.PersonFieldsApiFp)(this.configuration).deletePersonFields(requestParameters.ids).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Returns data about a specific person field.
     * @summary Get one person field
     * @param {PersonFieldsApiGetPersonFieldRequest} requestParameters Request parameters.

     * @throws {RequiredError}
     * @memberof PersonFieldsApi
     */
    getPersonField(requestParameters) {
        return (0, exports.PersonFieldsApiFp)(this.configuration).getPersonField(requestParameters.id).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Returns data about all person fields.<br>If a company uses the [Campaigns product](https://pipedrive.readme.io/docs/campaigns-in-pipedrive-api), then this endpoint will also return the `data.marketing_status` field.
     * @summary Get all person fields
     * @param {PersonFieldsApiGetPersonFieldsRequest} requestParameters Request parameters.

     * @throws {RequiredError}
     * @memberof PersonFieldsApi
     */
    getPersonFields(requestParameters = {}) {
        return (0, exports.PersonFieldsApiFp)(this.configuration).getPersonFields(requestParameters.start, requestParameters.limit).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Updates a person field. For more information, see the tutorial for <a href=\" https://pipedrive.readme.io/docs/updating-custom-field-value \" target=\"_blank\" rel=\"noopener noreferrer\">updating custom fields\' values</a>.
     * @summary Update a person field
     * @param {PersonFieldsApiUpdatePersonFieldRequest} requestParameters Request parameters.

     * @throws {RequiredError}
     * @memberof PersonFieldsApi
     */
    updatePersonField(requestParameters) {
        return (0, exports.PersonFieldsApiFp)(this.configuration).updatePersonField(requestParameters.id, requestParameters.UpdateFieldRequest).then((request) => request(this.axios, this.basePath));
    }
}
exports.PersonFieldsApi = PersonFieldsApi;
