// pipedrive/dist/versions/v1/api/product-fields-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductFieldsApi = exports.ProductFieldsApiFactory = exports.ProductFieldsApiFp = exports.ProductFieldsApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * ProductFieldsApi - axios parameter creator
 * @export
 */
const ProductFieldsApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Adds a new product field. For more information, see the tutorial for <a href=\"https://pipedrive.readme.io/docs/adding-a-new-custom-field\" target=\"_blank\" rel=\"noopener noreferrer\">adding a new custom field</a>.
         * @summary Add a new product field
         * @param {ProductFieldAllOf} [ProductFieldAllOf]

         * @throws {RequiredError}
         */
        addProductField: (ProductFieldAllOf) => __awaiter(this, void 0, void 0, function* () {
            const localVarPath = `/productFields`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign({ method: 'POST' }, baseOptions);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication api_key required
            yield (0, common_1.setApiKeyToObject)(localVarQueryParameter, "api_token", configuration);
            // authentication oauth2 required
            // oauth required
            yield (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["products:full"], configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(ProductFieldAllOf, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         * Marks a product field as deleted. For more information, see the tutorial for <a href=\"https://pipedrive.readme.io/docs/deleting-a-custom-field\" target=\"_blank\" rel=\"noopener noreferrer\">deleting a custom field</a>.
         * @summary Delete a product field
         * @param {number} id The ID of the product field

         * @throws {RequiredError}
         */
        deleteProductField: (id) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('deleteProductField', 'id', id);
            const localVarPath = `/productFields/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign({ method: 'DELETE' }, baseOptions);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication api_key required
            yield (0, common_1.setApiKeyToObject)(localVarQueryParameter, "api_token", configuration);
            // authentication oauth2 required
            // oauth required
            yield (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["products:full"], configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         * Marks multiple fields as deleted.
         * @summary Delete multiple product fields in bulk
         * @param {string} ids The comma-separated field IDs to delete

         * @throws {RequiredError}
         */
        deleteProductFields: (ids) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'ids' is not null or undefined
            (0, common_1.assertParamExists)('deleteProductFields', 'ids', ids);
            const localVarPath = `/productFields`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign({ method: 'DELETE' }, baseOptions);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication api_key required
            yield (0, common_1.setApiKeyToObject)(localVarQueryParameter, "api_token", configuration);
            // authentication oauth2 required
            // oauth required
            yield (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["products:full"], configuration);
            if (ids !== undefined) {
                localVarQueryParameter['ids'] = ids;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         * Returns data about a specific product field.
         * @summary Get one product field
         * @param {number} id The ID of the product field

         * @throws {RequiredError}
         */
        getProductField: (id) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('getProductField', 'id', id);
            const localVarPath = `/productFields/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign({ method: 'GET' }, baseOptions);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication api_key required
            yield (0, common_1.setApiKeyToObject)(localVarQueryParameter, "api_token", configuration);
            // authentication oauth2 required
            // oauth required
            yield (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["products:read", "products:full"], configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         * Returns data about all product fields.
         * @summary Get all product fields
         * @param {number} [start] Pagination start
         * @param {number} [limit] Items shown per page

         * @throws {RequiredError}
         */
        getProductFields: (start, limit) => __awaiter(this, void 0, void 0, function* () {
            const localVarPath = `/productFields`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign({ method: 'GET' }, baseOptions);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication api_key required
            yield (0, common_1.setApiKeyToObject)(localVarQueryParameter, "api_token", configuration);
            // authentication oauth2 required
            // oauth required
            yield (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["products:read", "products:full"], configuration);
            if (start !== undefined) {
                localVarQueryParameter['start'] = start;
            }
            if (limit !== undefined) {
                localVarQueryParameter['limit'] = limit;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         * Updates a product field. For more information, see the tutorial for <a href=\" https://pipedrive.readme.io/docs/updating-custom-field-value \" target=\"_blank\" rel=\"noopener noreferrer\">updating custom fields\' values</a>.
         * @summary Update a product field
         * @param {number} id The ID of the product field
         * @param {UpdateProductFieldResponse} [UpdateProductFieldResponse]

         * @throws {RequiredError}
         */
        updateProductField: (id, UpdateProductFieldResponse) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('updateProductField', 'id', id);
            const localVarPath = `/productFields/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign({ method: 'PUT' }, baseOptions);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication api_key required
            yield (0, common_1.setApiKeyToObject)(localVarQueryParameter, "api_token", configuration);
            // authentication oauth2 required
            // oauth required
            yield (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["products:full"], configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(UpdateProductFieldResponse, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
    };
};
exports.ProductFieldsApiAxiosParamCreator = ProductFieldsApiAxiosParamCreator;
/**
 * ProductFieldsApi - functional programming interface
 * @export
 */
const ProductFieldsApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.ProductFieldsApiAxiosParamCreator)(configuration);
    return {
        /**
         * Adds a new product field. For more information, see the tutorial for <a href=\"https://pipedrive.readme.io/docs/adding-a-new-custom-field\" target=\"_blank\" rel=\"noopener noreferrer\">adding a new custom field</a>.
         * @summary Add a new product field
         * @param {ProductFieldAllOf} [ProductFieldAllOf]

         * @throws {RequiredError}
         */
        addProductField(ProductFieldAllOf) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.addProductField(ProductFieldAllOf);
                return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         * Marks a product field as deleted. For more information, see the tutorial for <a href=\"https://pipedrive.readme.io/docs/deleting-a-custom-field\" target=\"_blank\" rel=\"noopener noreferrer\">deleting a custom field</a>.
         * @summary Delete a product field
         * @param {number} id The ID of the product field

         * @throws {RequiredError}
         */
        deleteProductField(id) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.deleteProductField(id);
                return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         * Marks multiple fields as deleted.
         * @summary Delete multiple product fields in bulk
         * @param {string} ids The comma-separated field IDs to delete

         * @throws {RequiredError}
         */
        deleteProductFields(ids) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.deleteProductFields(ids);
                return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         * Returns data about a specific product field.
         * @summary Get one product field
         * @param {number} id The ID of the product field

         * @throws {RequiredError}
         */
        getProductField(id) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getProductField(id);
                return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         * Returns data about all product fields.
         * @summary Get all product fields
         * @param {number} [start] Pagination start
         * @param {number} [limit] Items shown per page

         * @throws {RequiredError}
         */
        getProductFields(start, limit) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getProductFields(start, limit);
                return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         * Updates a product field. For more information, see the tutorial for <a href=\" https://pipedrive.readme.io/docs/updating-custom-field-value \" target=\"_blank\" rel=\"noopener noreferrer\">updating custom fields\' values</a>.
         * @summary Update a product field
         * @param {number} id The ID of the product field
         * @param {UpdateProductFieldResponse} [UpdateProductFieldResponse]

         * @throws {RequiredError}
         */
        updateProductField(id, UpdateProductFieldResponse) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.updateProductField(id, UpdateProductFieldResponse);
                return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
    };
};
exports.ProductFieldsApiFp = ProductFieldsApiFp;
/**
 * ProductFieldsApi - factory interface
 * @export
 */
const ProductFieldsApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.ProductFieldsApiFp)(configuration);
    return {
        /**
         * Adds a new product field. For more information, see the tutorial for <a href=\"https://pipedrive.readme.io/docs/adding-a-new-custom-field\" target=\"_blank\" rel=\"noopener noreferrer\">adding a new custom field</a>.
         * @summary Add a new product field
         * @param {ProductFieldsApiAddProductFieldRequest} requestParameters Request parameters.

         * @throws {RequiredError}
         */
        addProductField(requestParameters = {}) {
            return localVarFp.addProductField(requestParameters.ProductFieldAllOf).then((request) => request(axios, basePath));
        },
        /**
         * Marks a product field as deleted. For more information, see the tutorial for <a href=\"https://pipedrive.readme.io/docs/deleting-a-custom-field\" target=\"_blank\" rel=\"noopener noreferrer\">deleting a custom field</a>.
         * @summary Delete a product field
         * @param {ProductFieldsApiDeleteProductFieldRequest} requestParameters Request parameters.

         * @throws {RequiredError}
         */
        deleteProductField(requestParameters) {
            return localVarFp.deleteProductField(requestParameters.id).then((request) => request(axios, basePath));
        },
        /**
         * Marks multiple fields as deleted.
         * @summary Delete multiple product fields in bulk
         * @param {ProductFieldsApiDeleteProductFieldsRequest} requestParameters Request parameters.

         * @throws {RequiredError}
         */
        deleteProductFields(requestParameters) {
            return localVarFp.deleteProductFields(requestParameters.ids).then((request) => request(axios, basePath));
        },
        /**
         * Returns data about a specific product field.
         * @summary Get one product field
         * @param {ProductFieldsApiGetProductFieldRequest} requestParameters Request parameters.

         * @throws {RequiredError}
         */
        getProductField(requestParameters) {
            return localVarFp.getProductField(requestParameters.id).then((request) => request(axios, basePath));
        },
        /**
         * Returns data about all product fields.
         * @summary Get all product fields
         * @param {ProductFieldsApiGetProductFieldsRequest} requestParameters Request parameters.

         * @throws {RequiredError}
         */
        getProductFields(requestParameters = {}) {
            return localVarFp.getProductFields(requestParameters.start, requestParameters.limit).then((request) => request(axios, basePath));
        },
        /**
         * Updates a product field. For more information, see the tutorial for <a href=\" https://pipedrive.readme.io/docs/updating-custom-field-value \" target=\"_blank\" rel=\"noopener noreferrer\">updating custom fields\' values</a>.
         * @summary Update a product field
         * @param {ProductFieldsApiUpdateProductFieldRequest} requestParameters Request parameters.

         * @throws {RequiredError}
         */
        updateProductField(requestParameters) {
            return localVarFp.updateProductField(requestParameters.id, requestParameters.UpdateProductFieldResponse).then((request) => request(axios, basePath));
        },
    };
};
exports.ProductFieldsApiFactory = ProductFieldsApiFactory;
/**
 * ProductFieldsApi - object-oriented interface
 * @export
 * @class ProductFieldsApi
 * @extends {BaseAPI}
 */
class ProductFieldsApi extends base_1.BaseAPI {
    /**
     * Adds a new product field. For more information, see the tutorial for <a href=\"https://pipedrive.readme.io/docs/adding-a-new-custom-field\" target=\"_blank\" rel=\"noopener noreferrer\">adding a new custom field</a>.
     * @summary Add a new product field
     * @param {ProductFieldsApiAddProductFieldRequest} requestParameters Request parameters.

     * @throws {RequiredError}
     * @memberof ProductFieldsApi
     */
    addProductField(requestParameters = {}) {
        return (0, exports.ProductFieldsApiFp)(this.configuration).addProductField(requestParameters.ProductFieldAllOf).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Marks a product field as deleted. For more information, see the tutorial for <a href=\"https://pipedrive.readme.io/docs/deleting-a-custom-field\" target=\"_blank\" rel=\"noopener noreferrer\">deleting a custom field</a>.
     * @summary Delete a product field
     * @param {ProductFieldsApiDeleteProductFieldRequest} requestParameters Request parameters.

     * @throws {RequiredError}
     * @memberof ProductFieldsApi
     */
    deleteProductField(requestParameters) {
        return (0, exports.ProductFieldsApiFp)(this.configuration).deleteProductField(requestParameters.id).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Marks multiple fields as deleted.
     * @summary Delete multiple product fields in bulk
     * @param {ProductFieldsApiDeleteProductFieldsRequest} requestParameters Request parameters.

     * @throws {RequiredError}
     * @memberof ProductFieldsApi
     */
    deleteProductFields(requestParameters) {
        return (0, exports.ProductFieldsApiFp)(this.configuration).deleteProductFields(requestParameters.ids).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Returns data about a specific product field.
     * @summary Get one product field
     * @param {ProductFieldsApiGetProductFieldRequest} requestParameters Request parameters.

     * @throws {RequiredError}
     * @memberof ProductFieldsApi
     */
    getProductField(requestParameters) {
        return (0, exports.ProductFieldsApiFp)(this.configuration).getProductField(requestParameters.id).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Returns data about all product fields.
     * @summary Get all product fields
     * @param {ProductFieldsApiGetProductFieldsRequest} requestParameters Request parameters.

     * @throws {RequiredError}
     * @memberof ProductFieldsApi
     */
    getProductFields(requestParameters = {}) {
        return (0, exports.ProductFieldsApiFp)(this.configuration).getProductFields(requestParameters.start, requestParameters.limit).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Updates a product field. For more information, see the tutorial for <a href=\" https://pipedrive.readme.io/docs/updating-custom-field-value \" target=\"_blank\" rel=\"noopener noreferrer\">updating custom fields\' values</a>.
     * @summary Update a product field
     * @param {ProductFieldsApiUpdateProductFieldRequest} requestParameters Request parameters.

     * @throws {RequiredError}
     * @memberof ProductFieldsApi
     */
    updateProductField(requestParameters) {
        return (0, exports.ProductFieldsApiFp)(this.configuration).updateProductField(requestParameters.id, requestParameters.UpdateProductFieldResponse).then((request) => request(this.axios, this.basePath));
    }
}
exports.ProductFieldsApi = ProductFieldsApi;
