// pipedrive/dist/versions/v1/api/subscriptions-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubscriptionsApi = exports.SubscriptionsApiFactory = exports.SubscriptionsApiFp = exports.SubscriptionsApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * SubscriptionsApi - axios parameter creator
 * @export
 */
const SubscriptionsApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Adds a new recurring subscription.
         * @summary Add a recurring subscription
         * @param {AddRecurringSubscriptionRequest} [AddRecurringSubscriptionRequest]
         * @deprecated
         * @throws {RequiredError}
         */
        addRecurringSubscription: (AddRecurringSubscriptionRequest) => __awaiter(this, void 0, void 0, function* () {
            const localVarPath = `/subscriptions/recurring`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign({ method: 'POST' }, baseOptions);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication api_key required
            yield (0, common_1.setApiKeyToObject)(localVarQueryParameter, "api_token", configuration);
            // authentication oauth2 required
            // oauth required
            yield (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["deals:full"], configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(AddRecurringSubscriptionRequest, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         * Adds a new installment subscription.
         * @summary Add an installment subscription
         * @param {AddSubscriptionInstallmentRequest} [AddSubscriptionInstallmentRequest]
         * @deprecated
         * @throws {RequiredError}
         */
        addSubscriptionInstallment: (AddSubscriptionInstallmentRequest) => __awaiter(this, void 0, void 0, function* () {
            const localVarPath = `/subscriptions/installment`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign({ method: 'POST' }, baseOptions);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication api_key required
            yield (0, common_1.setApiKeyToObject)(localVarQueryParameter, "api_token", configuration);
            // authentication oauth2 required
            // oauth required
            yield (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["deals:full"], configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(AddSubscriptionInstallmentRequest, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         * Cancels a recurring subscription.
         * @summary Cancel a recurring subscription
         * @param {number} id The ID of the subscription
         * @param {CancelRecurringSubscriptionRequest} [CancelRecurringSubscriptionRequest]
         * @deprecated
         * @throws {RequiredError}
         */
        cancelRecurringSubscription: (id, CancelRecurringSubscriptionRequest) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('cancelRecurringSubscription', 'id', id);
            const localVarPath = `/subscriptions/recurring/{id}/cancel`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign({ method: 'PUT' }, baseOptions);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication api_key required
            yield (0, common_1.setApiKeyToObject)(localVarQueryParameter, "api_token", configuration);
            // authentication oauth2 required
            // oauth required
            yield (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["deals:full"], configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(CancelRecurringSubscriptionRequest, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         * Marks an installment or a recurring subscription as deleted.
         * @summary Delete a subscription
         * @param {number} id The ID of the subscription
         * @deprecated
         * @throws {RequiredError}
         */
        deleteSubscription: (id) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('deleteSubscription', 'id', id);
            const localVarPath = `/subscriptions/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign({ method: 'DELETE' }, baseOptions);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication api_key required
            yield (0, common_1.setApiKeyToObject)(localVarQueryParameter, "api_token", configuration);
            // authentication oauth2 required
            // oauth required
            yield (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["deals:full"], configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         * Returns details of an installment or a recurring subscription by the deal ID.
         * @summary Find subscription by deal
         * @param {number} dealId The ID of the deal
         * @deprecated
         * @throws {RequiredError}
         */
        findSubscriptionByDeal: (dealId) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'dealId' is not null or undefined
            (0, common_1.assertParamExists)('findSubscriptionByDeal', 'dealId', dealId);
            const localVarPath = `/subscriptions/find/{dealId}`
                .replace(`{${"dealId"}}`, encodeURIComponent(String(dealId)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign({ method: 'GET' }, baseOptions);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication api_key required
            yield (0, common_1.setApiKeyToObject)(localVarQueryParameter, "api_token", configuration);
            // authentication oauth2 required
            // oauth required
            yield (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["deals:read", "deals:full"], configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         * Returns details of an installment or a recurring subscription.
         * @summary Get details of a subscription
         * @param {number} id The ID of the subscription
         * @deprecated
         * @throws {RequiredError}
         */
        getSubscription: (id) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('getSubscription', 'id', id);
            const localVarPath = `/subscriptions/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign({ method: 'GET' }, baseOptions);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication api_key required
            yield (0, common_1.setApiKeyToObject)(localVarQueryParameter, "api_token", configuration);
            // authentication oauth2 required
            // oauth required
            yield (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["deals:read", "deals:full"], configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         * Returns all payments of an installment or recurring subscription.
         * @summary Get all payments of a subscription
         * @param {number} id The ID of the subscription
         * @deprecated
         * @throws {RequiredError}
         */
        getSubscriptionPayments: (id) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('getSubscriptionPayments', 'id', id);
            const localVarPath = `/subscriptions/{id}/payments`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign({ method: 'GET' }, baseOptions);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication api_key required
            yield (0, common_1.setApiKeyToObject)(localVarQueryParameter, "api_token", configuration);
            // authentication oauth2 required
            // oauth required
            yield (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["deals:read", "deals:full"], configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         * Updates a recurring subscription.
         * @summary Update a recurring subscription
         * @param {number} id The ID of the subscription
         * @param {UpdateRecurringSubscriptionRequest} [UpdateRecurringSubscriptionRequest]
         * @deprecated
         * @throws {RequiredError}
         */
        updateRecurringSubscription: (id, UpdateRecurringSubscriptionRequest) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('updateRecurringSubscription', 'id', id);
            const localVarPath = `/subscriptions/recurring/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign({ method: 'PUT' }, baseOptions);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication api_key required
            yield (0, common_1.setApiKeyToObject)(localVarQueryParameter, "api_token", configuration);
            // authentication oauth2 required
            // oauth required
            yield (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["deals:full"], configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(UpdateRecurringSubscriptionRequest, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         * Updates an installment subscription.
         * @summary Update an installment subscription
         * @param {number} id The ID of the subscription
         * @param {UpdateSubscriptionInstallmentRequest} [UpdateSubscriptionInstallmentRequest]
         * @deprecated
         * @throws {RequiredError}
         */
        updateSubscriptionInstallment: (id, UpdateSubscriptionInstallmentRequest) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('updateSubscriptionInstallment', 'id', id);
            const localVarPath = `/subscriptions/installment/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign({ method: 'PUT' }, baseOptions);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication api_key required
            yield (0, common_1.setApiKeyToObject)(localVarQueryParameter, "api_token", configuration);
            // authentication oauth2 required
            // oauth required
            yield (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["deals:full"], configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(UpdateSubscriptionInstallmentRequest, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
    };
};
exports.SubscriptionsApiAxiosParamCreator = SubscriptionsApiAxiosParamCreator;
/**
 * SubscriptionsApi - functional programming interface
 * @export
 */
const SubscriptionsApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.SubscriptionsApiAxiosParamCreator)(configuration);
    return {
        /**
         * Adds a new recurring subscription.
         * @summary Add a recurring subscription
         * @param {AddRecurringSubscriptionRequest} [AddRecurringSubscriptionRequest]
         * @deprecated
         * @throws {RequiredError}
         */
        addRecurringSubscription(AddRecurringSubscriptionRequest) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.addRecurringSubscription(AddRecurringSubscriptionRequest);
                return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         * Adds a new installment subscription.
         * @summary Add an installment subscription
         * @param {AddSubscriptionInstallmentRequest} [AddSubscriptionInstallmentRequest]
         * @deprecated
         * @throws {RequiredError}
         */
        addSubscriptionInstallment(AddSubscriptionInstallmentRequest) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.addSubscriptionInstallment(AddSubscriptionInstallmentRequest);
                return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         * Cancels a recurring subscription.
         * @summary Cancel a recurring subscription
         * @param {number} id The ID of the subscription
         * @param {CancelRecurringSubscriptionRequest} [CancelRecurringSubscriptionRequest]
         * @deprecated
         * @throws {RequiredError}
         */
        cancelRecurringSubscription(id, CancelRecurringSubscriptionRequest) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.cancelRecurringSubscription(id, CancelRecurringSubscriptionRequest);
                return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         * Marks an installment or a recurring subscription as deleted.
         * @summary Delete a subscription
         * @param {number} id The ID of the subscription
         * @deprecated
         * @throws {RequiredError}
         */
        deleteSubscription(id) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.deleteSubscription(id);
                return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         * Returns details of an installment or a recurring subscription by the deal ID.
         * @summary Find subscription by deal
         * @param {number} dealId The ID of the deal
         * @deprecated
         * @throws {RequiredError}
         */
        findSubscriptionByDeal(dealId) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.findSubscriptionByDeal(dealId);
                return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         * Returns details of an installment or a recurring subscription.
         * @summary Get details of a subscription
         * @param {number} id The ID of the subscription
         * @deprecated
         * @throws {RequiredError}
         */
        getSubscription(id) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getSubscription(id);
                return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         * Returns all payments of an installment or recurring subscription.
         * @summary Get all payments of a subscription
         * @param {number} id The ID of the subscription
         * @deprecated
         * @throws {RequiredError}
         */
        getSubscriptionPayments(id) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getSubscriptionPayments(id);
                return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         * Updates a recurring subscription.
         * @summary Update a recurring subscription
         * @param {number} id The ID of the subscription
         * @param {UpdateRecurringSubscriptionRequest} [UpdateRecurringSubscriptionRequest]
         * @deprecated
         * @throws {RequiredError}
         */
        updateRecurringSubscription(id, UpdateRecurringSubscriptionRequest) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.updateRecurringSubscription(id, UpdateRecurringSubscriptionRequest);
                return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
        /**
         * Updates an installment subscription.
         * @summary Update an installment subscription
         * @param {number} id The ID of the subscription
         * @param {UpdateSubscriptionInstallmentRequest} [UpdateSubscriptionInstallmentRequest]
         * @deprecated
         * @throws {RequiredError}
         */
        updateSubscriptionInstallment(id, UpdateSubscriptionInstallmentRequest) {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.updateSubscriptionInstallment(id, UpdateSubscriptionInstallmentRequest);
                return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
    };
};
exports.SubscriptionsApiFp = SubscriptionsApiFp;
/**
 * SubscriptionsApi - factory interface
 * @export
 */
const SubscriptionsApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.SubscriptionsApiFp)(configuration);
    return {
        /**
         * Adds a new recurring subscription.
         * @summary Add a recurring subscription
         * @param {SubscriptionsApiAddRecurringSubscriptionRequest} requestParameters Request parameters.
         * @deprecated
         * @throws {RequiredError}
         */
        addRecurringSubscription(requestParameters = {}) {
            return localVarFp.addRecurringSubscription(requestParameters.AddRecurringSubscriptionRequest).then((request) => request(axios, basePath));
        },
        /**
         * Adds a new installment subscription.
         * @summary Add an installment subscription
         * @param {SubscriptionsApiAddSubscriptionInstallmentRequest} requestParameters Request parameters.
         * @deprecated
         * @throws {RequiredError}
         */
        addSubscriptionInstallment(requestParameters = {}) {
            return localVarFp.addSubscriptionInstallment(requestParameters.AddSubscriptionInstallmentRequest).then((request) => request(axios, basePath));
        },
        /**
         * Cancels a recurring subscription.
         * @summary Cancel a recurring subscription
         * @param {SubscriptionsApiCancelRecurringSubscriptionRequest} requestParameters Request parameters.
         * @deprecated
         * @throws {RequiredError}
         */
        cancelRecurringSubscription(requestParameters) {
            return localVarFp.cancelRecurringSubscription(requestParameters.id, requestParameters.CancelRecurringSubscriptionRequest).then((request) => request(axios, basePath));
        },
        /**
         * Marks an installment or a recurring subscription as deleted.
         * @summary Delete a subscription
         * @param {SubscriptionsApiDeleteSubscriptionRequest} requestParameters Request parameters.
         * @deprecated
         * @throws {RequiredError}
         */
        deleteSubscription(requestParameters) {
            return localVarFp.deleteSubscription(requestParameters.id).then((request) => request(axios, basePath));
        },
        /**
         * Returns details of an installment or a recurring subscription by the deal ID.
         * @summary Find subscription by deal
         * @param {SubscriptionsApiFindSubscriptionByDealRequest} requestParameters Request parameters.
         * @deprecated
         * @throws {RequiredError}
         */
        findSubscriptionByDeal(requestParameters) {
            return localVarFp.findSubscriptionByDeal(requestParameters.dealId).then((request) => request(axios, basePath));
        },
        /**
         * Returns details of an installment or a recurring subscription.
         * @summary Get details of a subscription
         * @param {SubscriptionsApiGetSubscriptionRequest} requestParameters Request parameters.
         * @deprecated
         * @throws {RequiredError}
         */
        getSubscription(requestParameters) {
            return localVarFp.getSubscription(requestParameters.id).then((request) => request(axios, basePath));
        },
        /**
         * Returns all payments of an installment or recurring subscription.
         * @summary Get all payments of a subscription
         * @param {SubscriptionsApiGetSubscriptionPaymentsRequest} requestParameters Request parameters.
         * @deprecated
         * @throws {RequiredError}
         */
        getSubscriptionPayments(requestParameters) {
            return localVarFp.getSubscriptionPayments(requestParameters.id).then((request) => request(axios, basePath));
        },
        /**
         * Updates a recurring subscription.
         * @summary Update a recurring subscription
         * @param {SubscriptionsApiUpdateRecurringSubscriptionRequest} requestParameters Request parameters.
         * @deprecated
         * @throws {RequiredError}
         */
        updateRecurringSubscription(requestParameters) {
            return localVarFp.updateRecurringSubscription(requestParameters.id, requestParameters.UpdateRecurringSubscriptionRequest).then((request) => request(axios, basePath));
        },
        /**
         * Updates an installment subscription.
         * @summary Update an installment subscription
         * @param {SubscriptionsApiUpdateSubscriptionInstallmentRequest} requestParameters Request parameters.
         * @deprecated
         * @throws {RequiredError}
         */
        updateSubscriptionInstallment(requestParameters) {
            return localVarFp.updateSubscriptionInstallment(requestParameters.id, requestParameters.UpdateSubscriptionInstallmentRequest).then((request) => request(axios, basePath));
        },
    };
};
exports.SubscriptionsApiFactory = SubscriptionsApiFactory;
/**
 * SubscriptionsApi - object-oriented interface
 * @export
 * @class SubscriptionsApi
 * @extends {BaseAPI}
 */
class SubscriptionsApi extends base_1.BaseAPI {
    /**
     * Adds a new recurring subscription.
     * @summary Add a recurring subscription
     * @param {SubscriptionsApiAddRecurringSubscriptionRequest} requestParameters Request parameters.
     * @deprecated
     * @throws {RequiredError}
     * @memberof SubscriptionsApi
     */
    addRecurringSubscription(requestParameters = {}) {
        return (0, exports.SubscriptionsApiFp)(this.configuration).addRecurringSubscription(requestParameters.AddRecurringSubscriptionRequest).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Adds a new installment subscription.
     * @summary Add an installment subscription
     * @param {SubscriptionsApiAddSubscriptionInstallmentRequest} requestParameters Request parameters.
     * @deprecated
     * @throws {RequiredError}
     * @memberof SubscriptionsApi
     */
    addSubscriptionInstallment(requestParameters = {}) {
        return (0, exports.SubscriptionsApiFp)(this.configuration).addSubscriptionInstallment(requestParameters.AddSubscriptionInstallmentRequest).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Cancels a recurring subscription.
     * @summary Cancel a recurring subscription
     * @param {SubscriptionsApiCancelRecurringSubscriptionRequest} requestParameters Request parameters.
     * @deprecated
     * @throws {RequiredError}
     * @memberof SubscriptionsApi
     */
    cancelRecurringSubscription(requestParameters) {
        return (0, exports.SubscriptionsApiFp)(this.configuration).cancelRecurringSubscription(requestParameters.id, requestParameters.CancelRecurringSubscriptionRequest).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Marks an installment or a recurring subscription as deleted.
     * @summary Delete a subscription
     * @param {SubscriptionsApiDeleteSubscriptionRequest} requestParameters Request parameters.
     * @deprecated
     * @throws {RequiredError}
     * @memberof SubscriptionsApi
     */
    deleteSubscription(requestParameters) {
        return (0, exports.SubscriptionsApiFp)(this.configuration).deleteSubscription(requestParameters.id).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Returns details of an installment or a recurring subscription by the deal ID.
     * @summary Find subscription by deal
     * @param {SubscriptionsApiFindSubscriptionByDealRequest} requestParameters Request parameters.
     * @deprecated
     * @throws {RequiredError}
     * @memberof SubscriptionsApi
     */
    findSubscriptionByDeal(requestParameters) {
        return (0, exports.SubscriptionsApiFp)(this.configuration).findSubscriptionByDeal(requestParameters.dealId).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Returns details of an installment or a recurring subscription.
     * @summary Get details of a subscription
     * @param {SubscriptionsApiGetSubscriptionRequest} requestParameters Request parameters.
     * @deprecated
     * @throws {RequiredError}
     * @memberof SubscriptionsApi
     */
    getSubscription(requestParameters) {
        return (0, exports.SubscriptionsApiFp)(this.configuration).getSubscription(requestParameters.id).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Returns all payments of an installment or recurring subscription.
     * @summary Get all payments of a subscription
     * @param {SubscriptionsApiGetSubscriptionPaymentsRequest} requestParameters Request parameters.
     * @deprecated
     * @throws {RequiredError}
     * @memberof SubscriptionsApi
     */
    getSubscriptionPayments(requestParameters) {
        return (0, exports.SubscriptionsApiFp)(this.configuration).getSubscriptionPayments(requestParameters.id).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Updates a recurring subscription.
     * @summary Update a recurring subscription
     * @param {SubscriptionsApiUpdateRecurringSubscriptionRequest} requestParameters Request parameters.
     * @deprecated
     * @throws {RequiredError}
     * @memberof SubscriptionsApi
     */
    updateRecurringSubscription(requestParameters) {
        return (0, exports.SubscriptionsApiFp)(this.configuration).updateRecurringSubscription(requestParameters.id, requestParameters.UpdateRecurringSubscriptionRequest).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Updates an installment subscription.
     * @summary Update an installment subscription
     * @param {SubscriptionsApiUpdateSubscriptionInstallmentRequest} requestParameters Request parameters.
     * @deprecated
     * @throws {RequiredError}
     * @memberof SubscriptionsApi
     */
    updateSubscriptionInstallment(requestParameters) {
        return (0, exports.SubscriptionsApiFp)(this.configuration).updateSubscriptionInstallment(requestParameters.id, requestParameters.UpdateSubscriptionInstallmentRequest).then((request) => request(this.axios, this.basePath));
    }
}
exports.SubscriptionsApi = SubscriptionsApi;
