// pipedrive/dist/versions/v1/api/user-connections-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserConnectionsApi = exports.UserConnectionsApiFactory = exports.UserConnectionsApiFp = exports.UserConnectionsApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * UserConnectionsApi - axios parameter creator
 * @export
 */
const UserConnectionsApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Returns data about all connections for the authorized user.
         * @summary Get all user connections

         * @throws {RequiredError}
         */
        getUserConnections: () => __awaiter(this, void 0, void 0, function* () {
            const localVarPath = `/userConnections`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign({ method: 'GET' }, baseOptions);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication api_key required
            yield (0, common_1.setApiKeyToObject)(localVarQueryParameter, "api_token", configuration);
            // authentication oauth2 required
            // oauth required
            yield (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["base"], configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
    };
};
exports.UserConnectionsApiAxiosParamCreator = UserConnectionsApiAxiosParamCreator;
/**
 * UserConnectionsApi - functional programming interface
 * @export
 */
const UserConnectionsApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.UserConnectionsApiAxiosParamCreator)(configuration);
    return {
        /**
         * Returns data about all connections for the authorized user.
         * @summary Get all user connections

         * @throws {RequiredError}
         */
        getUserConnections() {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getUserConnections();
                return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
    };
};
exports.UserConnectionsApiFp = UserConnectionsApiFp;
/**
 * UserConnectionsApi - factory interface
 * @export
 */
const UserConnectionsApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.UserConnectionsApiFp)(configuration);
    return {
        /**
         * Returns data about all connections for the authorized user.
         * @summary Get all user connections

         * @throws {RequiredError}
         */
        getUserConnections() {
            return localVarFp.getUserConnections().then((request) => request(axios, basePath));
        },
    };
};
exports.UserConnectionsApiFactory = UserConnectionsApiFactory;
/**
 * UserConnectionsApi - object-oriented interface
 * @export
 * @class UserConnectionsApi
 * @extends {BaseAPI}
 */
class UserConnectionsApi extends base_1.BaseAPI {
    /**
     * Returns data about all connections for the authorized user.
     * @summary Get all user connections

     * @throws {RequiredError}
     * @memberof UserConnectionsApi
     */
    getUserConnections() {
        return (0, exports.UserConnectionsApiFp)(this.configuration).getUserConnections().then((request) => request(this.axios, this.basePath));
    }
}
exports.UserConnectionsApi = UserConnectionsApi;
