// pipedrive/dist/versions/v1/api/user-settings-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserSettingsApi = exports.UserSettingsApiFactory = exports.UserSettingsApiFp = exports.UserSettingsApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * UserSettingsApi - axios parameter creator
 * @export
 */
const UserSettingsApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Lists the settings of an authorized user. Example response contains a shortened list of settings.
         * @summary List settings of an authorized user

         * @throws {RequiredError}
         */
        getUserSettings: () => __awaiter(this, void 0, void 0, function* () {
            const localVarPath = `/userSettings`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign({ method: 'GET' }, baseOptions);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication api_key required
            yield (0, common_1.setApiKeyToObject)(localVarQueryParameter, "api_token", configuration);
            // authentication oauth2 required
            // oauth required
            yield (0, common_1.setOAuthToObject)(localVarHeaderParameter, "oauth2", ["base"], configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
    };
};
exports.UserSettingsApiAxiosParamCreator = UserSettingsApiAxiosParamCreator;
/**
 * UserSettingsApi - functional programming interface
 * @export
 */
const UserSettingsApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.UserSettingsApiAxiosParamCreator)(configuration);
    return {
        /**
         * Lists the settings of an authorized user. Example response contains a shortened list of settings.
         * @summary List settings of an authorized user

         * @throws {RequiredError}
         */
        getUserSettings() {
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getUserSettings();
                return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
            });
        },
    };
};
exports.UserSettingsApiFp = UserSettingsApiFp;
/**
 * UserSettingsApi - factory interface
 * @export
 */
const UserSettingsApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.UserSettingsApiFp)(configuration);
    return {
        /**
         * Lists the settings of an authorized user. Example response contains a shortened list of settings.
         * @summary List settings of an authorized user

         * @throws {RequiredError}
         */
        getUserSettings() {
            return localVarFp.getUserSettings().then((request) => request(axios, basePath));
        },
    };
};
exports.UserSettingsApiFactory = UserSettingsApiFactory;
/**
 * UserSettingsApi - object-oriented interface
 * @export
 * @class UserSettingsApi
 * @extends {BaseAPI}
 */
class UserSettingsApi extends base_1.BaseAPI {
    /**
     * Lists the settings of an authorized user. Example response contains a shortened list of settings.
     * @summary List settings of an authorized user

     * @throws {RequiredError}
     * @memberof UserSettingsApi
     */
    getUserSettings() {
        return (0, exports.UserSettingsApiFp)(this.configuration).getUserSettings().then((request) => request(this.axios, this.basePath));
    }
}
exports.UserSettingsApi = UserSettingsApi;
