// pipedrive/dist/versions/v1/models/add-webhook-request.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AddWebhookRequestVersionConst = exports.AddWebhookRequestEventObjectConst = exports.AddWebhookRequestEventActionConst = void 0;
exports.AddWebhookRequestEventActionConst = {
    create: 'create',
    change: 'change',
    delete: 'delete',
    Star: '*'
};
exports.AddWebhookRequestEventObjectConst = {
    activity: 'activity',
    deal: 'deal',
    lead: 'lead',
    note: 'note',
    organization: 'organization',
    person: 'person',
    pipeline: 'pipeline',
    product: 'product',
    stage: 'stage',
    user: 'user',
    Star: '*'
};
exports.AddWebhookRequestVersionConst = {
    _1_0: '1.0',
    _2_0: '2.0'
};
