// pipedrive/dist/versions/v1/models/add-note-request-all-of1.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AddNoteRequestAllOf1PinnedToPersonFlagConst = exports.AddNoteRequestAllOf1PinnedToOrganizationFlagConst = exports.AddNoteRequestAllOf1PinnedToDealFlagConst = exports.AddNoteRequestAllOf1PinnedToLeadFlagConst = void 0;
exports.AddNoteRequestAllOf1PinnedToLeadFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
exports.AddNoteRequestAllOf1PinnedToDealFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
exports.AddNoteRequestAllOf1PinnedToOrganizationFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
exports.AddNoteRequestAllOf1PinnedToPersonFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
