// pipedrive/dist/versions/v1/models/add-or-update-role-setting-request.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AddOrUpdateRoleSettingRequestValueConst = exports.AddOrUpdateRoleSettingRequestSettingKeyConst = void 0;
exports.AddOrUpdateRoleSettingRequestSettingKeyConst = {
    deal_default_visibility: 'deal_default_visibility',
    lead_default_visibility: 'lead_default_visibility',
    org_default_visibility: 'org_default_visibility',
    person_default_visibility: 'person_default_visibility',
    product_default_visibility: 'product_default_visibility'
};
exports.AddOrUpdateRoleSettingRequestValueConst = {
    NUMBER_1: 1,
    NUMBER_3: 3,
    NUMBER_5: 5,
    NUMBER_7: 7
};
