// pipedrive/dist/versions/v1/models/mail-message-data.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MailMessageDataHasRealAttachmentsFlagConst = exports.MailMessageDataHasInlineAttachmentsFlagConst = exports.MailMessageDataHasAttachmentsFlagConst = exports.MailMessageDataSmartBccFlagConst = exports.MailMessageDataSentFromPipedriveFlagConst = exports.MailMessageDataSentFlagConst = exports.MailMessageDataHasBodyFlagConst = exports.MailMessageDataDeletedFlagConst = exports.MailMessageDataSyncedFlagConst = exports.MailMessageDataDraftFlagConst = exports.MailMessageDataReadFlagConst = exports.MailMessageDataMailLinkTrackingEnabledFlagConst = exports.MailMessageDataMailTrackingStatusConst = void 0;
exports.MailMessageDataMailTrackingStatusConst = {
    opened: 'opened',
    not_opened: 'not opened'
};
exports.MailMessageDataMailLinkTrackingEnabledFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
exports.MailMessageDataReadFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
exports.MailMessageDataDraftFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
exports.MailMessageDataSyncedFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
exports.MailMessageDataDeletedFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
exports.MailMessageDataHasBodyFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
exports.MailMessageDataSentFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
exports.MailMessageDataSentFromPipedriveFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
exports.MailMessageDataSmartBccFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
exports.MailMessageDataHasAttachmentsFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
exports.MailMessageDataHasInlineAttachmentsFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
exports.MailMessageDataHasRealAttachmentsFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
