// pipedrive/dist/versions/v1/models/mail-thread-all-of1.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MailThreadAllOf1AllMessagesSentFlagConst = exports.MailThreadAllOf1FirstMessageToMeFlagConst = exports.MailThreadAllOf1ExternalDeletedFlagConst = exports.MailThreadAllOf1SharedFlagConst = exports.MailThreadAllOf1ArchivedFlagConst = exports.MailThreadAllOf1HasSentFlagConst = exports.MailThreadAllOf1HasDraftFlagConst = void 0;
exports.MailThreadAllOf1HasDraftFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
exports.MailThreadAllOf1HasSentFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
exports.MailThreadAllOf1ArchivedFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
exports.MailThreadAllOf1SharedFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
exports.MailThreadAllOf1ExternalDeletedFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
exports.MailThreadAllOf1FirstMessageToMeFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
exports.MailThreadAllOf1AllMessagesSentFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
