// pipedrive/dist/versions/v1/models/mail-thread-all-of.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MailThreadAllOfMailLinkTrackingEnabledFlagConst = exports.MailThreadAllOfSmartBccFlagConst = exports.MailThreadAllOfSyncedFlagConst = exports.MailThreadAllOfDeletedFlagConst = exports.MailThreadAllOfHasRealAttachmentsFlagConst = exports.MailThreadAllOfHasInlineAttachmentsFlagConst = exports.MailThreadAllOfHasAttachmentsFlagConst = exports.MailThreadAllOfReadFlagConst = void 0;
exports.MailThreadAllOfReadFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
exports.MailThreadAllOfHasAttachmentsFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
exports.MailThreadAllOfHasInlineAttachmentsFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
exports.MailThreadAllOfHasRealAttachmentsFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
exports.MailThreadAllOfDeletedFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
exports.MailThreadAllOfSyncedFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
exports.MailThreadAllOfSmartBccFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
exports.MailThreadAllOfMailLinkTrackingEnabledFlagConst = {
    NUMBER_0: 0,
    NUMBER_1: 1
};
